packages/web/src/pages/comment-history/components/desktop/CommentHistoryPage.tsx

```diff
@@ -0,0 +1,221 @@
+import Page from 'components/page/Page'
+import { useProfileParams } from 'pages/profile-page/useProfileParams'
+import { fullCommentHistoryPage } from 'utils/route'
+
+import { CommentText } from './CommentText'
+
+const messages = {
+  by: ' by ',
+  noComments: 'No comments',
+  view: 'View'
+}
+
+const UserComment = ({ commentId }: { commentId: number }) => {
+  const res = useGetCommentById(commentId)
+  const comment = res.data as Comment
+  const navigate = useNavigate()
+
+  const {
+    id,
+    entityId,
+    message,
+    createdAt,
+    userId,
+    isEdited,
+    mentions = [],
+    reactCount,
+    isCurrentUserReacted
+  } = comment
+
+  // @ts-expect-error - the id is a number
+  const { status } = useGetUserById({ id: userId })
+  const { data: track } = useGetTrackById({ id: decodeHashId(entityId) })
+  const isLoadingUser = status === Status.LOADING
+  const createdAtDate = useMemo(
+    () => dayjs.utc(createdAt).toDate(),
+    [createdAt]
+  )
+
+  const goToTrackPage = useCallback(() => {
+    if (track) {
+      navigate(track.permalink)
+    }
+  }, [track, navigate])
+
+  if (!comment) return null
+
+  return (
+    <Flex w='100%' gap='l' pv='m'>
+      <Box>
+        {/* @ts-expect-error - the id is a number */}
+        <Avatar userId={userId} size='medium' popover alignSelf='flex-start' />
+      </Box>
+      <Flex column w='100%' gap='s' alignItems='flex-start'>
+        <Flex column gap='xs' w='100%'>
+          <Flex>
+            <Text variant='body' size='s'>
+              {track ? (
+                <>
+                  <TrackLink isActive trackId={track?.track_id} />
+                  {messages.by}
+                  <UserLink isActive userId={track?.owner_id} />
+                </>
+              ) : (
+                <Skeleton w={180} h={20} />
+              )}
+            </Text>
+          </Flex>
+          <Flex column>
+            <Flex justifyContent='space-between'>
+              <Flex gap='s' alignItems='center'>
+                {isLoadingUser ? <Skeleton w={80} h={18} /> : null}
+                {userId !== undefined ? (
+                  <UserLink
+                    // @ts-expect-error - the id is a number
+                    userId={userId}
+                    popover
+                    size='l'
+                    strength='strong'
+                  />
+                ) : null}
+                <Flex gap='xs' alignItems='flex-end' h='100%'>
+                  <Timestamp time={createdAtDate} />
+                </Flex>
+              </Flex>
+            </Flex>
+            <CommentText
+              isEdited={isEdited}
+              isPreview={false}
+              mentions={mentions}
+              // @ts-expect-error - the id is a number
+              commentId={id}
+            >
+              {message}
+            </CommentText>
+          </Flex>
+        </Flex>
+        <Flex gap='l' alignItems='center'>
+          <Flex alignItems='center' gap='xs'>
```

Comment:
shouldn't this entire container take us to track page? what you think

Reply:
yea either works, but i guess it's if you expect clicking between or the number to navigate you. I'll update 👍